extensions/warp-ipfs/tests/community.rs

```diff
@@ -51,6 +51,19 @@ mod test {
         })
         .await
     }
+    async fn next_raygun_event(
+        stream: &mut RayGunEventStream,
+        timeout: Duration,
+    ) -> Result<RayGunEventKind, std::io::Error> {
+        crate::common::timeout(timeout, async {
+            loop {
+                if let Some(event) = stream.next().await {
+                    return event;
+                }
+            }
+        })
+        .await
+    }
```

Comment:
we could probably simplify this by doing


```rust
    async fn next_event<S: Stream + Unpin>(
        stream: &mut S,
        timeout: Duration,
    ) -> Result<S::Item, std::io::Error> {
        crate::common::timeout(timeout, async {
            loop {
                if let Some(event) = stream.next().await {
                    return event;
                }
            }
        })
        .await
    }
 ```
 
 That way we dont need separate functions for each stream

Reply:
We could probably ditch the loop as well since it would be expected that the stream wont terminate so we could also remove the and expect the next event. Maybe doing 

```rust
    async fn next_event<S: Stream + Unpin>(
        stream: &mut S,
        timeout: Duration,
    ) -> Result<S::Item, std::io::Error> {
        crate::common::timeout(
            timeout,
            stream
                .next()
                .map(|item| item.expect("stream not to terminate")),
        )
        .await
    }
```

Reply:
done 👍